FS19/ML/Zusammenfassung_ML.tex

```diff
@@ -1488,4 +1491,193 @@ \subsubsection{k-fold Cross Validation}
 	\item Pick the model $M-K$ with the lowest estimated generalisation error and retrain that model on the entirety of the training set $S$.
 \end{enumerate}
 
+\section{Clustering and Association Rules}
+
+\subsection{k-Means Algorithm}
+
+The k-Means algorithm is an algorithm that automatically identifies cluster centers. One has to specify the amount of clusters one wants before it can start though. The algorithm functions as follows:
+
+\begin{enumerate}
+	\item Input: Number of clusters $k>0$ and data points $x_1,...,x_n \in \mathbb{R}^m$
+	\item Randomly choose $k$ cluster centers $\mu_1,...,\mu_k \in \mathbb{R}^m$
+	\item Repeat until convergence
+	\begin{enumerate}
+		\item Assign each data point $x_i$ to its neares cluster center $\mu_j$
+		\begin{equation*}
+			c_i = \underset{j\in\{1,...,k\}}{\text{argmin}}\ \| x_i - \mu_j \|^2
+		\end{equation*}
+		\item  Update each cluster center to the mean of all assigned data points
+		\begin{equation*}
+			\mu_j := \frac{\sum_{i:c_i=j} x_i}{|\{i:c_i = j\}|}
+		\end{equation*}
+	\end{enumerate}
+\end{enumerate}
+
+\subsubsection{Addendum}
```

Comment:
What is this (OwO)?

Reply:
![grafik](https://user-images.githubusercontent.com/7289947/59541022-ea497c80-8eff-11e9-9816-457f384b324a.png)